backend/adapter_resource_test.go

```diff
@@ -65,25 +71,79 @@ func TestCallResource(t *testing.T) {
 		var actualRequestData map[string]interface{}
-		err = json.Unmarshal(res.Body, &actualRequestData)
+		err = json.Unmarshal(req.Body, &actualRequestData)
 		require.NoError(t, err)
 		require.Equal(t, data, actualRequestData)
 		require.Equal(t, int64(2), handler.actualReq.PluginConfig.OrgID)
 		require.Equal(t, "my-plugin", handler.actualReq.PluginConfig.PluginID)
 		require.Equal(t, "my-type", handler.actualReq.PluginConfig.PluginType)
 
 		// response
-		require.NotNil(t, res)
-		require.Equal(t, http.StatusOK, int(res.Code))
-		require.Contains(t, res.Headers, "X-Header-Out-1")
-		require.Equal(t, []string{"D", "E"}, res.Headers["X-Header-Out-1"].Values)
-		require.Contains(t, res.Headers, "X-Header-Out-2")
-		require.Equal(t, []string{"F"}, res.Headers["X-Header-Out-2"].Values)
+		require.Len(t, testSender.respMessages, 1)
+		resp := testSender.respMessages[0]
+		require.NotNil(t, resp)
+		require.Equal(t, http.StatusOK, int(resp.Code))
+		require.Contains(t, resp.Headers, "X-Header-Out-1")
+		require.Equal(t, []string{"D", "E"}, resp.Headers["X-Header-Out-1"].Values)
+		require.Contains(t, resp.Headers, "X-Header-Out-2")
+		require.Equal(t, []string{"F"}, resp.Headers["X-Header-Out-2"].Values)
 		var actualResponseData map[string]interface{}
-		err = json.Unmarshal(res.Body, &actualResponseData)
+		err = json.Unmarshal(resp.Body, &actualResponseData)
 		require.NoError(t, err)
 		require.Equal(t, data, actualResponseData)
 	})
+
+	t.Run("When call resource handler set should result in expected streaming response", func(t *testing.T) {
+		handler := &testCallResourceStreamHandler{
+			responseStatus: http.StatusOK,
+			responseHeaders: map[string][]string{
+				"X-Header-Out-1": []string{"D", "E"},
+				"X-Header-Out-2": []string{"F"},
+			},
+			responseMessages: [][]byte{
+				[]byte("hello"),
+				[]byte("world"),
+				[]byte("over and out"),
+			},
+		}
+		testSender := newTestCallResourceServer()
+		adapter := &sdkAdapter{
+			CallResourceHandler: handler,
+		}
+		req := &pluginv2.CallResource_Request{
+			Config: &pluginv2.PluginConfig{
+				OrgId:      2,
+				PluginId:   "my-plugin",
+				PluginType: "my-type",
+			},
+			Path:    "some/path",
+			Method:  http.MethodGet,
+			Url:     "plugins/test-plugin/resources/some/path?test=1",
+			Headers: map[string]*pluginv2.CallResource_StringList{},
+		}
+		err := adapter.CallResource(req, testSender)
+
+		require.NoError(t, err)
+
+		// response
+		require.Len(t, testSender.respMessages, 3)
+		resp1 := testSender.respMessages[0]
+		require.NotNil(t, resp1)
+		require.Equal(t, http.StatusOK, int(resp1.Code))
+		require.Contains(t, resp1.Headers, "X-Header-Out-1")
+		require.Equal(t, []string{"D", "E"}, resp1.Headers["X-Header-Out-1"].Values)
+		require.Contains(t, resp1.Headers, "X-Header-Out-2")
+		require.Equal(t, []string{"F"}, resp1.Headers["X-Header-Out-2"].Values)
+		require.Equal(t, "hello", string(resp1.Body))
+
+		resp2 := testSender.respMessages[1]
+		require.NotNil(t, resp2)
```

Comment:
This test is kind of pointless since test the content on next line :)
I've done this in plenty of cases as well but I think we should remove the `NotNil` check if we check the content later. It's just more text without any value.

Reply:
It's to guard against "invalid memory address or nil pointer dereference". Instead of getting a panic. Or what am I missing here?

Reply:
no your right. I thought the assertion framework would yield a better output in those cases. My bad.